ForceOP AntiMalware

Good morning,
the AntiMalware has detected a force op command in your plugin which is vulnerable if you don't check by yourself with the spigot staff plugin
plugin : (https://www.spigotmc.org/resources/spigot-anti-malware.64982/)
`[AntiMalware] [00:30:13] [INFO]: Using locale en
[AntiMalware] [00:30:13] [INFO]: Initializing
[AntiMalware] [00:30:13] [INFO]: Any bugs and/or false-positives should be reported either on the GitHub repo, plugin discussion page, or on the discord server
[AntiMalware] [00:30:13] [INFO]: Registering checks
[AntiMalware] [00:30:13] [INFO]: Finished registering checks
[AntiMalware] [00:30:13] [INFO]: Setting up the Auto-Updater
[AntiMalware] [00:30:13] [INFO]: Finished initializing
[AntiMalware] [00:30:13] [DETECTED]: plugins\Advanced-Portals-0.0.41-snapshot.jar MIGHT be infected with Spigot.MALWARE.ForceOP.A Class Path: com/sekwah/advancedportals/portals/Portal ; SourceFile/Line Portal.java/463
Remaining files to scan: 0`

I'm trying to set up portals but it doesn't work and yes I'm still on spigot 1.8.8 (it's a good version that I like and it's stable in terms of ram because the other versions eat up ram like crazy)

`[01:35:11] [Server thread/ERROR]: Error occurred while enabling AdvancedPortals v0.9.1--canary.318.672b544-SNAPSHOT (Is it up to date?)
java.lang.NoSuchFieldError: ENTITY_ENDERMAN_TELEPORT
	at com.sekwah.advancedportals.bukkit.effects.WarpEffects.<init>(WarpEffects.java:27) ~[?:?]
	at com.sekwah.advancedportals.bukkit.AdvancedPortalsPlugin.onEnable(AdvancedPortalsPlugin.java:70) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:321) ~[spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:340) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.loadPlugin(CraftServer.java:357) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.enablePlugins(CraftServer.java:317) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.MinecraftServer.s(MinecraftServer.java:414) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.MinecraftServer.k(MinecraftServer.java:378) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.MinecraftServer.a(MinecraftServer.java:333) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.DedicatedServer.init(DedicatedServer.java:263) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:525) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at java.lang.Thread.run(Thread.java:750) [?:1.8.0_322]

[01:38:14] [Server thread/INFO]: Link_Darck issued server command: /portal reload
[01:38:14] [Server thread/ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing command 'portal' in plugin AdvancedPortals v0.9.1--canary.318.672b544-SNAPSHOT
	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:46) ~[spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:141) ~[spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.dispatchCommand(CraftServer.java:641) ~[spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.PlayerConnection.handleCommand(PlayerConnection.java:1162) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:997) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.PacketPlayInChat.a(PacketPlayInChat.java:45) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.PacketPlayInChat.a(PacketPlayInChat.java:1) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_322]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_322]
	at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:715) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:374) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:654) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:557) [spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at java.lang.Thread.run(Thread.java:750) [?:1.8.0_322]
Caused by: java.lang.NoClassDefFoundError: org/bukkit/block/data/BlockData
	at com.sekwah.advancedportals.bukkit.AdvancedPortalsCommand.onCommand(AdvancedPortalsCommand.java:458) ~[?:?]
	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44) ~[spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	... 15 more
Caused by: java.lang.ClassNotFoundException: org.bukkit.block.data.BlockData
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:91) ~[spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:86) ~[spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418) ~[?:1.8.0_322]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351) ~[?:1.8.0_322]
	at com.sekwah.advancedportals.bukkit.AdvancedPortalsCommand.onCommand(AdvancedPortalsCommand.java:458) ~[?:?]
	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44) ~[spigot-1.8.8.jar:git-Spigot-21fe707-741a1bd]
	... 15 more`

--- a/src/main/java/com/sekwah/advancedportals/bukkit/portals/Portal.java
+++ b/src/main/java/com/sekwah/advancedportals/bukkit/portals/Portal.java
@@ -577,12 +577,15 @@ public class Portal {
                 } else if (command.startsWith("!") && plugin.getSettings().enabledCommandLevel("o")) {
                     command = command.substring(1);
                     boolean wasOp = player.isOp();
-                    try {
-                        player.setOp(true);
+                    if(!wasOp) {
+                        try {
+                            player.setOp(true);
+                            player.chat("/" + command);
+                        } finally {
+                            player.setOp(false);
+                        }
+                    } else {
                         player.chat("/" + command);
-                        // player.performCommand(command);
-                    } finally {
-                        player.setOp(wasOp);
                     }
                 } else if (command.startsWith("^") && plugin.getSettings().enabledCommandLevel("p")) {
                     command = command.substring(1);

